t/50reverse-proxy-https.t

```diff
@@ -30,16 +38,17 @@ hosts:
       "/no-verify":
         proxy.reverse.url: https://127.0.0.1:$upstream_port
         proxy.ssl.verify-peer: OFF
-      "/wikipedia":
-        proxy.reverse.url: https://en.wikipedia.org/wiki/Main_Page
+      "/examp1e":
+        proxy.reverse.url: https://localhost.examp1e.net:$examp1e_server->{tls_port}
+        proxy.ssl.verify-peer: OFF
 EOT
     run_with_curl($server, sub {
         my ($proto, $port, $curl) = @_;
         my $resp = `$curl --silent --dump-header /dev/stderr --max-redirs 0 $proto://127.0.0.1:$port/verify/ 2>&1 > /dev/null`;
         like $resp, qr{^HTTP/[^ ]* 502\s}is;
         $resp = `$curl --silent --dump-header /dev/stderr --max-redirs 0 $proto://127.0.0.1:$port/no-verify/ 2>&1 > /dev/null`;
         unlike $resp, qr{^HTTP/[^ ]* 502\s}is;
-        $resp = `$curl --silent --dump-header /dev/stderr --max-redirs 0 $proto://127.0.0.1:$port/wikipedia/ 2>&1 > /dev/null`;
+        $resp = `$curl --silent --dump-header /dev/stderr --max-redirs 0 $proto://127.0.0.1:$port/examp1e/ 2>&1 > /dev/null`;
```

Comment:
These three tests test three behaviors:
* When connecting to an HTTPS server that does not provide a valid certificate, h2o running as a reverse proxy with `proxy.ssl.verify-peer` set to `ON` returns 502.
* When connecting to an HTTPS server that does not provide a valid certificate, h2o running as a reverse proxy with `proxy.ssl.verify-peer` set to `OFF` skips peer verification and forwards the request-response.
* When connecting to an HTTP server that provides a valid certificate, h2o running as a reverse proxy with `proxy.ssl.verify-peer` set to `ON` forwards the request-response.

With the proposed change, we'd no longer cover the third case.

Fortunately, after we added these tests, we have created our own test CA and have reissued certs stored under examples/h2o. It is now possible to use our own test CA for testing the third behavior.

Therefore, the correct thing to do here would be to change the third entry of the test configuration to use `https://127.0.0.1:$upstream_port` with [proxy.ssl.ca.file](https://h2o.examp1e.net/configure/proxy_directives.html#proxy.ssl.cafile) set to `misc/test-ca/root/ca.crt`, with `proxy.ssl.verify-peer` set to `ON`, and check that a 200 response is returned.

Reply:
Thank you for your review.  I think you mean to continue to use `https://localhost.examp1e.net:$examp1e_server->{tls_port}` but correct the ssl options?  Because in this case the upstream is the `examp1e_server` instance of `h2o`.  Please check the changes in https://github.com/h2o/h2o/pull/2790/commits/73ec1e92fa0be56575d0e0677ec90ce72b7c0e88.